wallet-new/src/Cardano/Wallet/Launcher.hs

```diff
@@ -0,0 +1,290 @@
+                     allocateNodeResources, bracketNodeResources,
+                     loggerBracket, releaseNodeResources, runNode, runNodeReal,
+                     withConfigurations)
+import           Pos.Ssc.Types (SscParams)
+import           Pos.Txp (txpGlobalSettings)
+import           Pos.Util.CompileInfo (HasCompileInfo, withCompileInfo)
+import           Pos.Util.Util (logException)
+import           Pos.Wallet.WalletMode (WalletMempoolExt)
+import           Pos.Wallet.Web.Account (getSKById)
+import           Pos.Wallet.Web.Mode (WalletWebMode, WalletWebModeContext (..))
+import           Pos.Wallet.Web.Server.Launcher (bracketWalletWS,
+                     bracketWalletWebDB)
+import           Pos.Wallet.Web.Server.Runner (runWRealMode)
+import           Pos.Wallet.Web.State (askWalletDB, askWalletSnapshot,
+                     flushWalletStorage, getWalletAddresses)
+import           Pos.Wallet.Web.State.Acidic (closeState, openState)
+import           Pos.Wallet.Web.Tracking (eskToWalletDecrCredentials,
+                     syncWallet)
+import           Pos.Web.Types (TlsParams (..))
+import           Pos.Worker.Update (updateTriggerWorker)
+import           Pos.WorkMode (RealModeContext (..))
+
+import qualified Cardano.Wallet.Kernel.Keystore as Keystore
+import qualified Cardano.Wallet.Kernel.Mode as Kernel.Mode
+import qualified Cardano.Wallet.Server.Plugins as Plugins
+
+
+-- | start a node from node arguments
+startCoreNode
+    :: CommonNodeArgs
+    -> NodeArgs
+    -> LoggingParams
+    -> IO ()
+startCoreNode cArgs nArgs lArgs = do
+    let blPath = AssetLockPath <$> cnaAssetLockPath cArgs
+
+    loggerBracket lArgs . logException (lpDefaultName lArgs) $
+        withCompileInfo $ withConfigurations blPath (getConfigOpts cArgs) $ \_ pm -> do
+            (params, Just ssc) <- getNodeParams (lpDefaultName lArgs) cArgs nArgs
+            runNodeReal pm params ssc [updateTriggerWorker]
+
+
+-- | Runs an wallet node plus its wallet backend API.
+startWalletNode
+    :: NodeArgs
+    -> WalletStartupOptions
+    -> LoggingParams
+    -> IO ()
+startWalletNode nArgs wOpts lArgs = do
+    let cArgs  = wsoNodeArgs wOpts
+    let blPath = AssetLockPath <$> cnaAssetLockPath cArgs
+
+    loggerBracket lArgs . logException (lpDefaultName lArgs) $ do
+        logInfo "[Attention] Software is built with the wallet backend"
+        withCompileInfo $ withConfigurations blPath (getConfigOpts cArgs) $ \ntpConfig pm -> do
+            (nodeParams, Just sscParams) <- getNodeParams (lpDefaultName lArgs) cArgs nArgs
+            printInfoOnStart cArgs ntpConfig
+            logInfo "Wallet is enabled!"
+            case wsoWalletBackendParams wOpts of
+              WalletLegacy legacyParams ->
+                actionWithWallet pm sscParams nodeParams ntpConfig legacyParams
+              WalletNew newParams ->
+                actionWithNewWallet pm sscParams nodeParams newParams
```

Comment:
Note that, in the _theory_, wallet node are intended to be _edge nodes_ with lower functionalities (they shouldn't validate blocks for instance). In practice, it seems that they're merely _core nodes_ exposing an extra wallet API.